Code cov - add cpp unit tests to code cov

LakshmiKumar23 opened the pull request:
Added cpp unit tests and external source reader
Code cov up by 23.18%.
Total coverage is now 43.59%
https://app.codecov.io/gh/ROCm/rocAL/pull/319
Ready to be merged
Will add audio and video tests as a separate PR

LakshmiKumar23 committed:
attempt to add more code cov

LakshmiKumar23 committed:
fix to run the test

LakshmiKumar23 committed:
modify permissions

LakshmiKumar23 committed:
fix for path for MIVisionX-data

rrawther reviewed (changes requested):
@LakshmiKumar23 to test and make sure it works

samjwu committed:
Update common.groovy

escape $ with \

samjwu committed:
Update common.groovy

update dir after running tests

rocal/build/test contains the profraw files in rawdata/*.profraw

LakshmiKumar23 committed:
path corrections for  generating rawprof

LakshmiKumar23 committed:
Merge branch 'lk/code_cov' of https://www.github.com/LakshmiKumar23/rocAL into lk/code_cov

LakshmiKumar23 committed:
path corrections for  generating rawprof

LakshmiKumar23 committed:
add external source reader test

LakshmiKumar23 committed:
return back to build diretcory

LakshmiKumar23 committed:
return to parent folder
